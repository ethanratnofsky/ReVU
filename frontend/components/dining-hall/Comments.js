import { FlatList, Text, View } from 'react-native';

import commentsStyles from './CommentsStyles';

import { ANONYMOUS_NAMES } from '../../constants';

const Comment = ({ content, timestamp}) => {
    const author = ANONYMOUS_NAMES[Math.floor(Math.random() * ANONYMOUS_NAMES.length)];
    const time = new Date(timestamp).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

    return (
        <View style={commentsStyles.commentContainer} onStartShouldSetResponder={() => true}>
            <View style={commentsStyles.commentHeader}>
                <Text style={commentsStyles.commentAuthor}>{author}</Text>
                <Text style={commentsStyles.commentTime}>{time}</Text>
            </View>
            <Text style={commentsStyles.commentContent}>{content}</Text>
        </View>
    )
};

const Comments = ({ comments }) => {
    const renderComment = ({ item }) => {
        return <Comment {...item} />
    };

    comments = [
        {
            id: 1,
            userId: 1,
            diningHallId: 5,
            content: 'food was mid and dining hall is far away',
            timestamp: Date.now(),
        },
        {
            id: 2,
            userId: 2,
            diningHallId: 3,
            content: 'E',
            timestamp: Date.now(),
        },
        {
            id: 3,
            userId: 3,
            diningHallId: 2,
            content: 'This is a really, really, really, really, really, really, really, really, really, really, really, really, really, really, really long comment about Rothschild',
            timestamp: Date.now(),
        }
    ]

    return (
        <View style={commentsStyles.container}>
            <Text style={commentsStyles.commentsHeader}>Comments â€¢ {comments.length}</Text>
            <View style={commentsStyles.divider} />
            <FlatList
                style={commentsStyles.commentsContainer}
                data={comments}
                renderItem={renderComment}
                keyExtractor={comment => comment._id}
            />
        </View>
    )
};

export default Comments;
