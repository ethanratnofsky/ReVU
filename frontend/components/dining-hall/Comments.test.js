import React from 'react';
const renderer = require('react-test-renderer');
import Comments from './Comments';

describe('<Comments />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Comments />).toJSON();
    expect(tree.children.length).toBe(3);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<Comments />).toJSON();
    expect(tree).toMatchSnapshot();
  });