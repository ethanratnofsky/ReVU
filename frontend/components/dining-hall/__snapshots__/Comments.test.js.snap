// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "LibreCaslonText_700Bold",
        "fontSize": 16,
      }
    }
  >
    Comments â€¢ 
    3
  </Text>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#D9D9D9",
        "height": 1,
        "marginHorizontal": 30,
        "marginVertical": 10,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "content": "food was mid and dining hall is far away",
          "diningHallId": 5,
          "id": 1,
          "timestamp": 1668579823824,
          "userId": 1,
        },
        Object {
          "content": "E",
          "diningHallId": 3,
          "id": 2,
          "timestamp": 1668579823824,
          "userId": 2,
        },
        Object {
          "content": "This is a really, really, really, really, really, really, really, really, really, really, really, really, really, really, really long comment about Rothschild",
          "diningHallId": 2,
          "id": 3,
          "timestamp": 1668579823824,
          "userId": 3,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "marginHorizontal": 15,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#222",
              "borderRadius": 3,
              "margin": 2,
              "padding": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontWeight": "bold",
                }
              }
            >
              Anonymous Dragon
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 11,
                }
              }
            >
              12:23 AM
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
          >
            food was mid and dining hall is far away
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#222",
              "borderRadius": 3,
              "margin": 2,
              "padding": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontWeight": "bold",
                }
              }
            >
              Anonymous Hippo
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 11,
                }
              }
            >
              12:23 AM
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
          >
            E
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#222",
              "borderRadius": 3,
              "margin": 2,
              "padding": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 3,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontWeight": "bold",
                }
              }
            >
              Anonymous Tomato
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 11,
                }
              }
            >
              12:23 AM
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
              }
            }
          >
            This is a really, really, really, really, really, really, really, really, really, really, really, really, really, really, really long comment about Rothschild
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
